c7n/resources/s3.py

```diff
@@ -375,6 +377,244 @@ def get_permissions(self):
+                        bad_bucket_lifecycle = False
+
+                        if rule['Status'] == 'Disabled':
+                            bad_bucket_lifecycle = True
+
+                        # Handle lifecycles created using new Amazon web interface
+                        if 'Filter' in rule:
+                            if rule['Filter']['Prefix'] != prefix:
+                                bad_bucket_lifecycle = True
+                        else:
+                            if rule['Prefix'] != prefix:
+                                bad_bucket_lifecycle = True
+
+                        if 'Transitions' in rule:
+                            # Can only have STANDARD_IA and GLACIER transitions
+                            if (len(rule['Transitions']) < 2) and\
+                                    ia_days_in_policy and\
+                                    glacier_days_in_policy:
+                                bad_bucket_lifecycle = True
+                            else:
+                                for t in rule['Transitions']:
+                                    if (t['StorageClass'] == 'STANDARD_IA'):
+                                        if ia_days_in_policy and (t['Days'] != ia_days_in_policy):
+                                            bad_bucket_lifecycle = True
+
+                                    if (t['StorageClass'] == 'GLACIER'):
+                                        if glacier_days_in_policy and\
+                                                (t['Days'] != glacier_days_in_policy):
+                                            bad_bucket_lifecycle = True
+
+                        elif ia_days_in_policy or glacier_days_in_policy:
+                            bad_bucket_lifecycle = True
+
+                        if 'Expiration' in rule and 'Days' in rule['Expiration']:
+                            if isinstance(deletes_objects, bool) and (deletes_objects is False):
+                                bad_bucket_lifecycle = True
+                            elif delete_days_in_policy and\
+                                    rule['Expiration']['Days'] != delete_days_in_policy:
+                                bad_bucket_lifecycle = True
+                        elif (deletes_objects is True) or delete_days_in_policy:
+                            bad_bucket_lifecycle = True
+
+                        if 'AbortIncompleteMultipartUpload' in rule:
+                            multipart_days_in_rule =\
+                                rule['AbortIncompleteMultipartUpload']['DaysAfterInitiation']
+                            if multipart_days_in_policy and\
+                                    (multipart_days_in_rule != multipart_days_in_policy):
+                                bad_bucket_lifecycle = True
+                        elif multipart_days_in_policy:
+                            bad_bucket_lifecycle = True
+
+                if not bad_bucket_lifecycle:
+                    results.append(b)
```

Comment:
I definitely see where there is unneeded complexity here. I have implemented your recommendation.